fuzz/fuzz_targets/compress_then_decompress_chunked.rs

```diff
@@ -0,0 +1,185 @@
+};
+
+use libfuzzer_sys::fuzz_target;
+
+fn compress_c(data: &[u8], compression_level: u8, work_factor: u8) -> Vec<u8> {
+    // compress the data with the stock C bzip2
+
+    // output buffer for compression, will get resized later if needed
+    let mut output = vec![0u8; 1024];
+
+    let mut stream = libbz2_rs_sys::bz_stream {
+        next_in: data.as_ptr() as *mut _,
+        avail_in: data.len() as _,
+        total_in_lo32: 0,
+        total_in_hi32: 0,
+        next_out: output.as_mut_ptr() as *mut _,
+        avail_out: output.len() as _,
+        total_out_lo32: 0,
+        total_out_hi32: 0,
+        state: std::ptr::null_mut(),
+        bzalloc: None,
+        bzfree: None,
+        opaque: std::ptr::null_mut(),
+    };
+
+    unsafe {
+        let err = libbz2_rs_sys::BZ2_bzCompressInit(
+            &mut stream,
+            compression_level.into(),
+            0,
+            work_factor.into(),
+        );
+        // init should always succeed
+        assert_eq!(err, BZ_OK);
+    };
+
+    let error = loop {
+        match unsafe { libbz2_rs_sys::BZ2_bzCompress(&mut stream, BZ_FINISH) } {
+            BZ_FINISH_OK => {
+                let used = output.len() - stream.avail_out as usize;
+                // The output buffer is full, resize it
+                let add_space: u32 = Ord::max(1024, output.len().try_into().unwrap());
+                output.resize(output.len() + add_space as usize, 0);
+
+                // If resize() reallocates, it may have moved in memory
+                stream.next_out = output.as_mut_ptr().cast::<i8>().wrapping_add(used);
+                stream.avail_out += add_space;
+
+                continue;
+            }
+            BZ_STREAM_END => {
+                break BZ_OK;
+            }
+            ret => {
+                break ret;
+            }
+        }
+    };
+
+    // compression should always succeed
+    assert_eq!(error, BZ_OK);
+
+    // truncate the output buffer down to the actual number of compressed bytes
+    output.truncate(
+        ((u64::from(stream.total_out_hi32) << 32) + u64::from(stream.total_out_lo32))
+            .try_into()
+            .unwrap(),
+    );
+
+    unsafe {
+        // cleanup, should always succeed
+        let err = libbz2_rs_sys::BZ2_bzCompressEnd(&mut stream);
+        assert_eq!(err, BZ_OK);
+    }
+
+    output
+}
+
+fuzz_target!(|input: (&[u8], usize, u8, u8)| {
+    let (fuzzer_data, chunk_size, compression_decider, work_factor_decider) = input;
+
+    // let the fuzzer pick a value from 1 to 9 (inclusive)
+    // use modulo to ensure this always maps to a valid number
+    let compression_level: u8 = (compression_decider % 9) + 1;
+
+    // valid values 0 to 250 (inclusive)
+    // use modulo to ensure this always maps to a valid number
+    let work_factor = work_factor_decider % 251;
+
+    if chunk_size == 0 {
+        // we can't iterate over chunks of size 0 byte, exit early
+        return;
+    }
```

Comment:
In theory this is where `Corpus::Reject` would be useful right? Though probably overkill in this specific case.